src/debugging/netSdk/NetSdkDebugHelper.ts

```diff
@@ -38,14 +46,70 @@ export class NetSdkDebugHelper extends NetCoreDebugHelper {
         return configurations;
     }
 
+    public async afterResolveDebugConfiguration(context: DockerDebugContext, debugConfiguration: DockerDebugConfiguration): Promise<void> {
+        const runTasks: Task[] = await tasks.fetchTasks({ type: NetSdkRunTaskType }) || [];
+        const runTaskFirstEntry = runTasks.find(t => t.source === NetSdkRunTaskType);
+        if (!runTaskFirstEntry) {
+            throw Error(l10n.t('No .NET SDK run task found '));
+        }
+        const taskExecution = await tasks.executeTask(runTaskFirstEntry);
```

Comment:
You can execute a task you define here, rather than having to match a pre-existing task. See: https://github.com/microsoft/vscode-extension-samples/blob/668632196f378491a65240463a16ce09629fe6c8/l10n-sample/src/extension.ts#L16 for an example with a generic shell task.

The idea would be to hardcode the appProject here rather than having to rely on mementos/state in the task itself to retrieve the target project.